spec/value_inspector_spec.rb

```diff
@@ -61,9 +66,154 @@
     let(:value) { [{ id: 1, contents: "some text" }, { id: 2 }] }
     it "is an array containing the schema" do
       expect(schema).to eq(
+        class: "Array",
+        items: [
+          {
+            class: "Hash",
+            properties: [
+              { name: :id, class: "Integer" },
+              { name: :contents, class: "String" },
+            ],
+          },
+          {
+            class: "Hash",
+            properties: [ { name: :id, class: "Integer" } ]
+          }
+        ],
+      )
+    end
+
+    describe "max depth" do
+      let(:max_depth) { 0 }
+      it "exceeds max_depth to describe element types when inspecting arrays" do
+        expect(schema).to eq(
+          class: "Array",
+          items: [ { class: "Hash" } ]
+        )
+      end
+    end
+  end
+
+  describe "Nested arrays" do
+    let(:value) { [ [[[1]], [[2]]], [[[3]], [[4]]] ] }
+    let(:max_depth) { 1 }
+    it "exceeds max depth to get type information" do
+      expect(schema).to eq(
+        class: "Array",
+        items: [
+          {
+            class: "Array",
+            items: [
+              class: "Array",
+              items: [
+                {
+                  class: "Array",
+                  items: [
+                    { class: "Integer" }
+                  ]
+                }
+              ]
+            ]
+          }
+        ]
+      )
+    end
+
+    describe "with Hashes" do
+      let(:max_depth) { 2 }
+      let(:value) do
+        [
+          {
+            id: 0,
+            children: [
+              { id: 1, children: [ { id: 2 } ] },
+              { id: 3, children: [ { id: 4 } ] }
+            ]
+          }
+        ]
+      end
+
+      it "won't exceed max_depth when inspecting a Hash" do
+        expect(schema).to eq(
+          class: "Array",
+          items: [
+            {
+              class: "Hash",
+              properties: [
+                { name: :id, class: "Integer" },
+                { name: :children, class: "Array", items: [ { class: "Hash" } ] }
```

Comment:
Is it valid to not have any schema on this hash?

Reply:
This is essentially the same as the following, which would be valid:
```yml
type: array
items: object
```